lib/commands/context.js

```diff
@@ -0,0 +1,472 @@
+};
+
+extensions.onPageChange = async function (pageArray) {
+  logger.debug(`Remote debugger notified us of a new page listing: ${JSON.stringify(pageArray)}`);
+  if (this.selectingNewPage) {
+    logger.debug('We are in the middle of selecting a page, ignoring');
+    return;
+  }
+  let newIds = [];
+  let newPages = [];
+  let keyId = null;
+  for (let page of pageArray) {
+    let id = page.id.toString();
+    newIds.push(id);
+    if (page.isKey) {
+      keyId = id;
+    }
+    if (!_.contains(this.contexts, id)) {
+      newPages.push(id);
+      this.contexts.push(id);
+    }
+  }
+
+  let newPage = null;
+  if (this.curContext === null) {
+    logger.debug('We do not appear to have window set yet, ignoring');
+  } else if (newPages.length) {
+    logger.debug(`We have new pages, going to select page '${newPages[0]}'`);
+    newPage = newPages[0];
+  } else if (!_.contains(newIds, this.curContext.toString())) {
+    logger.debug('New page listing from remote debugger does not contain ' +
+                 'current window; assuming it is closed');
+    if (keyId !== null) {
+      logger.debug(`Debugger already selected page '${keyId}', ` +
+                   `confirming that choice.`);
+    } else {
+      logger.error('Do not have our current window anymore, and there ' +
+                   'are not any more to load! Doing nothing...');
+      return;
+    }
+    this.curContext = keyId;
+    // this.remote.pageIdKey = parseInt(keyId, 10);
+    newPage = keyId;
+  } else {
+    let dirty = () => {
+      var item = (arr) => {
+        return _.filter(arr, function (obj) {
+          return obj.id === this.curContext;
+        }, this)[0];
+      };
+
+      return !_.isEqual(item(this.contexts), item(pageArray));
+    };
+
+    // If a window gets navigated to an anchor it doesn't always fire a page callback event
+    // Let's check if we wound up in such a situation.
+    if (dirty()) {
+      this.remote.pageLoad();
+    }
+
+    logger.debug('New page listing is same as old, doing nothing');
+  }
+
+  if (newPage !== null) {
+    this.selectingNewPage = true;
+    await this.remote.selectPage(parseInt(newPage, 10));
+    this.selectingNewPage = false;
+    this.curContext = newPage;
+  }
+  this.windowHandleCache = _.map(pageArray, this.massagePage);
+};
+
+extensions.getLatestWebviewContextForTitle = async function (titleRegex) {
+  let contexts = await this.getContextsAndViews();
+  let matchingCtx;
+  // _(contexts).each((ctx) => {
+  for (let ctx of contexts) {
+    if (ctx.view && (ctx.view.title || '').match(titleRegex)) {
+      if (ctx.view.url === 'about:blank') {
+        // in the case of Xcode  < 5 (i.e., iOS SDK Version less than 7)
+        // and in the case of iOS 7.1 in a webview (not in Safari)
+        // we can have the url be `about:blank`
+        if (parseFloat(this.iOSSDKVersion) < 7 ||
+            (this.caps.platformVersion === '7.1' && this.caps.app && this.caps.app.toLowerCase() !== 'safari')) {
```

Comment:
yeah, just following my bad example, we need to change them in all the files in the repo.